Fluid/Ast/FromStatement.cs

```diff
@@ -0,0 +1,114 @@
+    public sealed class FromStatement : Statement
+#pragma warning restore CA1001
+    {
+        public const string ViewExtension = ".liquid";
+
+        private readonly FluidParser _parser;
+        private volatile CachedTemplate _cachedTemplate;
+        private readonly SemaphoreSlim _semaphore = new SemaphoreSlim(1);
+    
+        public FromStatement(FluidParser parser, Expression path, List<string> functions = null)
+        {
+            _parser = parser;
+            Path = path;
+            Functions = functions;
+        }
+
+        public Expression Path { get; }
+        public List<String> Functions { get; }
+
+        public override async ValueTask<Completion> WriteToAsync(TextWriter writer, TextEncoder encoder, TemplateContext context)
+        {
+            var relativePath = (await Path.EvaluateAsync(context)).ToStringValue();
+            if (!relativePath.EndsWith(ViewExtension, StringComparison.OrdinalIgnoreCase))
+            {
+                relativePath += ViewExtension;
+            }
+
+            if (_cachedTemplate == null || !string.Equals(_cachedTemplate.Name, System.IO.Path.GetFileNameWithoutExtension(relativePath), StringComparison.Ordinal))
+            {
+                await _semaphore.WaitAsync();
+                try
+                {
+                    var fileProvider = context.Options.FileProvider;
+                    var fileInfo = fileProvider.GetFileInfo(relativePath);
+                    if (fileInfo == null || !fileInfo.Exists)
+                    {
+                        throw new FileNotFoundException(relativePath);
+                    }
+
+                    var content = "";
+
+                    using (var stream = fileInfo.CreateReadStream())
+                    using (var streamReader = new StreamReader(stream))
+                    {
+                        content = await streamReader.ReadToEndAsync();
+                    }
+
+                    if (!_parser.TryParse(content, out var template, out var errors))
+                    {
+                        throw new ParseException(errors);
+                    }
+
+                    var identifier = System.IO.Path.GetFileNameWithoutExtension(relativePath);
+                    _cachedTemplate = new CachedTemplate(template, identifier);
+                }
+                finally
+                {
+                    _semaphore.Release();
+                }
+            }
+
+            try
+            {
+                var parentScope = context.LocalScope;
+
+                context.EnterChildScope();
+                await _cachedTemplate.Template.RenderAsync(TextWriter.Null, encoder, context);
+
+                if (Functions != null)
+                {
+                    foreach (var functionName in Functions)
+                    {
+                        var value = context.LocalScope.GetValue(functionName);
+                        if (value.Type != FluidValues.Nil)
```

Comment:
```suggestion
                        if (value is FunctionValue)
```

Like the block below?